pageserver/src/tenant/timeline.rs

```diff
@@ -3161,7 +3190,7 @@ impl Timeline {
         }
     }
 
-    async fn flush_frozen_layers_and_wait(&self) -> anyhow::Result<()> {
+    async fn flush_frozen_layers_and_wait(&self, to_lsn: Lsn) -> anyhow::Result<()> {
```

Comment:
Comment added.

I think `frozen_lsn` is a bit confusing because we talk about frozen layers, but we don't have a concept of a "frozen LSN" -- the `to_lsn` here is really the last_record_lsn at the time we were called.  I've called it `last_record_lsn` to make that a bit more obvious.